src-test/org/etools/j1939_84/model/OBDModuleInformationTest.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2020. Equipment & Tool Institute
+ */
+package org.etools.j1939_84.model;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.etools.j1939_84.bus.j1939.packets.SupportedSPN;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Unit Test for the {@link OBDModuleInformation} class
+ *
+ * @author Marianne Schaefer (marianne.m.schaefer@gmail.com)
+ *
+ */
+public class OBDModuleInformationTest {
+
+    private static List<SupportedSPN> makeListOfSupportedSPNs(int[] data) {
+        List<SupportedSPN> supportedSpnsList = new ArrayList<>();
+        SupportedSPN supportedSpn = new SupportedSPN(data);
+        supportedSpnsList.add(supportedSpn);
+        return supportedSpnsList;
+    }
+
+    private OBDModuleInformation instance;
+
+    private OBDModuleInformation instance2;
+
+    @Before
+    public void setUp() throws Exception {
+
+        instance = new OBDModuleInformation(0);
+        instance.setObdCompliance((byte) 4);
+        instance.setSupportedSpns(makeListOfSupportedSPNs(new int[] { 4, 5, 6, 7, 8 }));
+
+        instance2 = new OBDModuleInformation(0);
+        instance2.setObdCompliance((byte) 4);
+        instance2.setSupportedSpns(makeListOfSupportedSPNs(new int[] { 4, 5, 6, 7, 8 }));
+
+    }
+
+    @Test
+    public void testEquals() {
+        assertFalse(instance.equals(new Object()));
+        assertFalse(instance.equals(null));
+        assertTrue(instance.equals(instance));
+        assertTrue(instance.equals(instance2));
+    }
+
+    @Test
+    public void testGetDataStreamSpns() {
+        assertTrue(instance.getDataStreamSpns().equals(instance2.getDataStreamSpns()));
+    }
+
+    @Test
+    public void testGetFreezeFrameSpns() {
+        assertTrue(instance.getFreezeFrameSpns().equals(instance2.getFreezeFrameSpns()));
+    }
+
+    @Test
+    public void testGetObdCompliance() {
+        assertEquals("ObdCompliance", (byte) 4, instance.getObdCompliance());
+        assertEquals("ObdCompliance", instance.getObdCompliance(), instance2.getObdCompliance());
```

Comment:
Using assertEquals here isn’t really right.  You’re asserting that when you call getObdCompliance on instance2, the expected value is what’s returned from the getObdCompliance on instance.  If you’re testing that they are the same, it should be assertTrue(instance.getObdCompliance() == instance2.getObdCompliance()). But that’s not really necessary here at all.  That’s handling when you’re testing equals